x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/sender/HttpRequestExecutorService.java

```diff
@@ -76,26 +83,71 @@ class HttpRequestExecutorService extends AbstractExecutorService {
     public void start() {
         try {
             while (running.get()) {
-                HttpTask task = queue.take();
-                if (task.shouldShutdown() || running.get() == false) {
-                    running.set(false);
-                    logger.debug(() -> format("Http executor service [%s] exiting", serviceName));
-                } else {
-                    executeTask(task);
-                }
+                handleTasks();
             }
         } catch (InterruptedException e) {
             Thread.currentThread().interrupt();
         } finally {
+            notifyRequestsOfShutdown();
```

Comment:
```suggestion
            running.set(false);
            notifyRequestsOfShutdown();
```

If interrupted or another error reject any more requests. This is a bit racy as a thread calling `send()` may have got past the isShutDown check but be blocked on or not got to the `queue.offer()` yet.


`PriorityProcessWorkerExecutorService` handles this case by with another `isShutdownCheck` after `queue.offer()` and calling `notifyRequestsOfShutdown()` if true. `notifyRequestsOfShutdown` has be a synchronised method as it may also be called by the thread executing `start()`. There may be a better way of handling this. 

The interrupt probably means this node is being shutdown but the request may have originated from another node that isn't being shutdown.

Reply:
Ah good point. I think I'm going to go with the same solution as `PriorityProcessWorkerExecutorService`. The only other option I can think of is to wrap the queue in a class that adds reentrant read/write locks. I tried that initially but it's more code and looks uglier 🤷‍♂️